### PR TITLE
Add hooks for using atmchange within testmods

### LINKED ISSUE #1781: testmods shell_commands file cannot run ./atmchange
testmods is run before case.setup, so atmchange and the data/*.xml files aren't available yet.

### PR DESCRIPTION
**Pull request by jgfouca**
Requires an env XML entry to buffer the change until it is ready to be applied.

Tested via:
`./create_test --no-build ERS_Ln22.ne30_ne30.F2010-SCREAMv1.mappy_gnu9.scream-scream_example_testmod_atmchange`

Confirmed that, in namelist_scream.xml, 
`<cubed_sphere_map type="integer">42</cubed_sphere_map>`

Fixes #1781

### CODE REVIEW

File: components/scream/cime_config/buildnml [hunk 1]
```diff
@@ -690,47 +697,6 @@ def create_input_files(caseroot, screamroot, rundir):
 
     return result
 
-###############################################################################
```
Comment (jgfouca) [inline components/scream/cime_config/buildnml:693]: This change is unrelated. I just felt that buildnml should be the first function you see as you scroll up from _main_func since buildnml is the "main" function.

**Review by ambrad — APPROVED** [review]
Thanks!

File: components/scream/cime_config/buildnml [hunk 2]
```diff
@@ -505,6 +505,13 @@ def create_raw_xml_file(case, caseroot):
         with open(raw_xml_file, "w") as fd:
             ET.ElementTree(raw_xml).write(fd, method='xml', encoding="unicode")
 
+    # Now that we have our namelist_scream.xml file, we can apply buffered
+    # atmchange requests.
+    atmchg_buffer = case.get_value("SCREAM_ATMCHANGE_BUFFER")
+    if atmchg_buffer:
+        run_cmd_no_fail("{}/atmchange {}".format(caseroot, atmchg_buffer))
```
Comment (bartgol) [inline components/scream/cime_config/buildnml:512]: Does this assume that there's only one atmchange cmd stored in the xml temp var?

Comment (bartgol) [reply components/scream/cime_config/buildnml:512]: I.e., does it work correctly with 
```
./xmlchange SCREAM_ATMCHANGE_BUFFER='cubed_sphere_map=42'
./xmlchange --append SCREAM_ATMCHANGE_BUFFER='cubed_sphere_map=24'
```
?

Comment (jgfouca) [reply components/scream/cime_config/buildnml:512]: @bartgol , I just tested it and it appears to work. xmlchange is even smart enough to add a whitespace when appending things:

```
% ./xmlquery SCREAM_ATMCHANGE_BUFFER
	SCREAM_ATMCHANGE_BUFFER: 
% ./xmlchange SCREAM_ATMCHANGE_BUFFER='foo=bar'
% ./xmlquery SCREAM_ATMCHANGE_BUFFER
	SCREAM_ATMCHANGE_BUFFER: foo=bar
% ./xmlchange SCREAM_ATMCHANGE_BUFFER='baz=bat' --append
% ./xmlquery SCREAM_ATMCHANGE_BUFFER
	SCREAM_ATMCHANGE_BUFFER: foo=bar baz=bat
```

Comment (jgfouca) [reply components/scream/cime_config/buildnml:512]: Oh, I see you wanted me to test case.setup, not xmlchange. That appears to work too:

```
% ./xmlchange SCREAM_ATMCHANGE_BUFFER='cubed_sphere_map=32'
% ./xmlchange SCREAM_ATMCHANGE_BUFFER='dt_remap_factor=42' --append
% ./xmlquery SCREAM_ATMCHANGE_BUFFER
	SCREAM_ATMCHANGE_BUFFER: cubed_sphere_map=32 dt_remap_factor=42
% /bin/rm namelist_scream.xml 
% ./case.setup
% grep -E '>(42|32)' namelist_scream.xml
    <cubed_sphere_map type="integer">32</cubed_sphere_map>
    <dt_remap_factor constraints="ge 1" type="integer">42</dt_remap_factor>
```
